app/models/channel.rb

```diff
@@ -167,6 +169,8 @@ def self.find_by_channel_identifier(channel_identifier)
         Rails.logger.info("Unable to find channel for channel identifier #{channel_identifier}")
         nil
     end
+  rescue NoMethodError => e
+    nil
```

Comment:
Would you use `LogException` rather than allowing a silent error?

Reply:
Can we use the nil safe navigation operator to prevent this?

Reply:
Can we use the nil safe navigation operator to prevent this?

Reply:
I do think we want to raise a RecordNotFound error if the record isn't found, but we could improve this by using the nil safe operator.  We could use do something like this:

```
result = YoutubeChannelDetails.where(youtube_channel_id: channel_details_type_identifier).
                       joins(:channel).                                joins(:channel).
                       where('channels.verified = true').first&.channel

...

raise ActiveRecord::RecordNotFound.new("Couldn't find Channel with 'channel_identifier'=#{channel_identifier}") if result.nil?
```

Reply:
Since this is largely not relevant to the PR, I'm going to revert to the original behavior and we can discuss in a different issue.